### DISCUSSION TITLE
Promise.all nested arrays in one response, undefined in 2nd response

### DISCUSSION THREAD

**Discussion by jimguistical** [nss-evening-cohort-14/evening-client-side#331] [Capstones!] [answered]
JIm Conner

Goal:
On click of my list "X" Button, I want to delete my whole List, then return one [ ] for listNameArray, one array for getSitesList, 
then I want to set the empty arrays in state in my useEffect, refresh the view onclick of list delete button. 

Describe Issue:
Issue is that my `Promise.all()` is returning a nested array on one response instead of each response having a separate array.  I think if I can get my Promise.all working, then I can `deleteListSItes()` run the  `getAllListData` and then set state in ListView useEffect.  I think I am missing a key piece related to manipulating data within my promise arrays before they get to the ListView.

User can create only one list right now, so I deleted everything:

```
const deleteListSites = (uid) => new Promise((resolve, reject) => {
  axios.delete(`${dbUrl}/resourcelist.json`)
    .then(() => getAllListData(uid).then((response) => resolve(response))
      .catch((error) => reject(error)));
});
```
Promise.all call: 
```
const getAllListData = (uid) => new Promise((resolve, reject) => {
  const getListNameArray = getListByListName(uid);
  const getSitesList = getList(uid);
  Promise.all([getListNameArray, getSitesList])
  // console.warn(getListNameArray, getSitesList)
    .then((listNameResponse, siteListResponse) => resolve(
      { getListNameArray: listNameResponse, getSitesList: siteListResponse }
    ))
    .catch((error) => reject(error));
});
```

Googled:
https://stackoverflow.com/questions/49744707/how-to-use-promise-all-on-array-of-promises-which-take-parameters
https://stackoverflow.com/questions/36094865/how-to-do-promise-all-for-array-of-array-of-promises
https://stackoverflow.com/questions/38427761/promise-all-with-nested-promise-all
https://forum.freecodecamp.org/t/how-to-map-a-subarray-or-nested-array/137761

Tickets reviewed: #306 #325

Links to files:
https://github.com/jim-conner/capstone-e14/blob/delete-allobjs/src/App/views/ListView.js
https://github.com/jim-conner/capstone-e14/blob/delete-allobjs/src/helpers/data/listData.js
https://github.com/jim-conner/capstone-e14/blob/delete-allobjs/src/helpers/data/siteData.js

console.warns below, showing getListNameArray response holds two arrays, and getSitesList response undefined:
![image](https://user-images.githubusercontent.com/76793747/122939145-fbbc8f00-d338-11eb-9e24-b76eff11b7d3.png)

**Comment by drteresavasquez** [comment, marked as answer]
You need brackets around the items in your .then. You're not restructuring the elements so they are coming back as an array of arrays.

**Reply by jimguistical** [reply]
I needed to destructure after the `Promise.all` runs so that I could get to the nested arrays, then I could pass the responses so that I can call them in my ListView and set state.  
Needed the destructure here: `.then(([listNameResponse, siteListResponse]) => resolve(
`
Code block:

```
const getAllListData = (uid) => new Promise((resolve, reject) => {
  const getListNameArray = getListByListName(uid);
  const getSitesList = getList(uid);
  Promise.all([getListNameArray, getSitesList])
    .then(([listNameResponse, siteListResponse]) => resolve(
      { getListNameArray: listNameResponse, getSitesList: siteListResponse }
    ))
    .catch((error) => reject(error));
});
```
